param/parameterized.py

```diff
@@ -1125,447 +1052,533 @@ def state_push(self):
+
+    unknown_value determines what to do where a representation cannot be
+    generated for something required to recreate the object. Such things
+    include non-parameter positional and keyword arguments, and certain
+    values of parameters (e.g. some random state objects).
 
-        qualifier = mod + '.'  if qualify else ''
-        arguments = arglist + keywords + (['**%s' % spec.varargs] if spec.varargs else [])
-        return qualifier + '%s(%s)' % (self.__class__.__name__,  (','+separator+prefix).join(arguments))
+    Supplying an unknown_value of None causes unrepresentable things
+    to be silently ignored. If unknown_value is a string, that
+    string will appear in place of any unrepresentable things. If
+    unknown_value is False, an Exception will be raised if an
+    unrepresentable value is encountered.
 
+    If supplied, imports should be a list, and it will be populated
+    with the set of imports required for the object and all of its
+    parameter values.
 
-    def __str__(self):
-        """Return a short representation of the name and class of this object."""
-        return "<%s %s>" % (self.__class__.__name__,self.name)
+    If qualify is True, the class's path will be included (e.g. "a.b.C()"),
+    otherwise only the class will appear ("C()").
 
+    Parameters will be separated by a comma only by default, but the
+    separator parameter allows an additional separator to be supplied
+    (e.g. a newline could be supplied to have each Parameter appear on a
+    separate line).
 
-    # CEBALERT: designed to avoid any processing unless the print
-    # level is high enough, but not all callers of message(),
-    # verbose(), debug(), etc are taking advantage of this. Need to
-    # document, and also check other ioam projects.
-    def __db_print(self,level,msg,*args,**kw):
-        """
-        Calls the logger returned by the get_logger() function,
-        prepending the result of calling dbprint_prefix() (if any).
+    NOTE: pprint will replace script_repr in a future version of
+    param, but is not yet a complete replacement for script_repr.
+    """
+    # CB: doc prefix & settings or realize they don't need to be
+    # passed around, etc.
+    # JLS: The settings argument is not used anywhere. To be removed
+    # in a separate PR.
+    if isinstance(val,type):
+        rep = type_script_repr(val,imports,prefix,settings)
 
-        See python's logging module for details.
-        """
-        if get_logger().isEnabledFor(level):
+    elif type(val) in script_repr_reg:
+        rep = script_repr_reg[type(val)](val,imports,prefix,settings)
 
-            if dbprint_prefix and callable(dbprint_prefix):
-                prefix=dbprint_prefix() # pylint: disable-msg=E1102
-            else:
-                prefix=""
+    # CEBALERT: remove with script_repr
+    elif hasattr(val,'script_repr'):
+        rep=val.script_repr(imports, prefix+"    ")
```

Comment:
@ceball, is this the bit of `pprint` that still requires `.script_repr()`?  Seems odd that a `.script_repr()` method would take precedence over `.pprint()` if `.pprint()` is meant to replace `.script_repr()`.